docs/DeveloperGuide.md

```diff
@@ -281,28 +320,105 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. User supplies an invalid file location to FriendDex
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. FriendDex shows an error message of the error.
+      
+      Use case resumes at step 2.
+    
+* 3b. User supplies a valid file location but wrong file format
 
+    * 3b1. FriendDex shows an error message of the error.
+      
       Use case resumes at step 2.
 
-*{More to be added}*
+**Use case: Add birthday to a person**
 
-### Non-Functional Requirements
+**MSS**
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1.  User requests to list persons
+2.  FriendDex shows a list of persons
+3.  User requests to add a birthday to a specific person in the list
+4.  FriendDex stores the birthday information for the person
+    
+    Use case ends
+    
+**Extensions**
 
-*{More to be added}*
+* 2a. The list is empty
+  
+  Use case ends
+  
+* 3a. The given date is invalid
+  
+    * 3a1. FriendDex shows an error message, and the accepted date format(s)
+      
+      Use case ends
 
-### Glossary
+**Use case: Create Friend Groups**
 
-* **Mainstream OS**: Windows, Linux, Unix, OS-X
```

Comment:
I think we should keep `Mainstream OS` in the glossary as mentioned in our NFRs

Reply:
Sounds good. Added the glossary back in.